src/core/manager.c

```diff
@@ -1389,6 +1389,29 @@ static unsigned manager_dispatch_gc_job_queue(Manager *m) {
         return n;
 }
 
+static int manager_ratelimit_check_and_queue(Unit *u, sd_event_time_handler_t cb) {
+        int r;
+
+        assert(u);
+        assert(cb);
+
+        if (ratelimit_below(&u->auto_start_stop_ratelimit))
+                return 1;
+
+        r = sd_event_add_time_relative(
+                        u->manager->event,
+                        &u->auto_start_stop_event_source,
+                        CLOCK_MONOTONIC,
+                        u->auto_start_stop_ratelimit.interval,
```

Comment:
please add:

```
static inline usec_t ratelimit_end(const RateLimit *rl) {
        assert(rl); 
        if (rl->begin == 0)
                return 0;
        return usec_add(rl->begin, rl->interval);
}
```

to ratelimit.h

and then use an absolute event source and use that for the absolute time to retry again.

this has the benefit that we'll try the instant the window is over, and not any later.